src/librustc/mir/interpret/allocation.rs

```diff
@@ -15,16 +15,17 @@ use std::borrow::Cow;
 
 #[derive(Clone, Debug, Eq, PartialEq, PartialOrd, Ord, Hash, RustcEncodable, RustcDecodable)]
 pub struct Allocation<Tag=(),Extra=()> {
-    /// The actual bytes of the allocation.
-    /// Note that the bytes of a pointer represent the offset of the pointer
-    pub bytes: Vec<u8>,
+    /// The byte allocation and definedness.
+    /// Bytes of fully undefined allocations are allocated lazily on potentially mutable changes.
```

Comment:
"potentially mutable changes"?

Reply:
It's always allocated within [`get_bytes_mut`](https://github.com/rust-lang/rust/blob/4ad12fb8806a9366ddba46a51dc42fc3c14e8f15/src/librustc/mir/interpret/allocation.rs#L275) but I hadn't realized that the method also marks the bytes as defined when I had written that comment. Good catch.